### PR TITLE
Add setting to allow disabling the Assistant

### PR DESCRIPTION
**Pull request by maxdeviant**
This PR adds a new `assistant.enabled` setting that controls whether the Zed Assistant is enabled.

Some users have requested the ability to disable the AI-related features in Zed if they don't use them. Changing `assistant.enabled` to `false` will hide the Assistant icon in the status bar (taking priority over the `assistant.button` setting) as well as filter out the `assistant:` actions.

The Assistant is enabled by default.

Release Notes:

- Added an `assistant.enabled` setting to control whether the Assistant is enabled.

### CODE REVIEW

File: crates/assistant/src/assistant_settings.rs [hunk 1]
```diff
@@ -201,42 +202,26 @@ impl AssistantSettingsContent {
...
-                            api_url: open_ai_api_url.clone(),
-                        }),
-                    }
+            AssistantSettingsContent::Legacy(settings) => AssistantSettingsContentV1 {
+                enabled: None,
+                button: settings.button,
+                dock: settings.dock,
+                default_width: settings.default_width,
+                default_height: settings.default_height,
+                provider: if let Some(open_ai_api_url) = settings.openai_api_url.as_ref() {
```
Comment (maxdeviant) [inline crates/assistant/src/assistant_settings.rs:211]: This part of the diff it just a refactoring, as I noticed that the branching only applied to the `provider` field.

**Comment by hferreiro** [comment]
I've just updated to 0.129.1 and the key `enabled` doesn't seem to work. Buttons and command palette actions are still shown.

**Comment by maxdeviant** [comment]
> I've just updated to 0.129.1 and the key `enabled` doesn't seem to work. Buttons and command palette actions are still shown.

I just tested in v0.129.2 and it's working for me.

Could you share the settings you're using?

**Comment by hferreiro** [comment]
```
  "assistant": {
     "enabled": false
  }
  ```

**Comment by maxdeviant** [comment]
> ```
>   "assistant": {
>      "enabled": false
>   }
> ```

We made some changes to the assistant settings that required a new version of the schema, so you need to specify that you're using that new version in order for `enabled` to work.

Could you try using these settings?

```json
"assistant": { "version": "1", "enabled": false },
```

**Comment by hferreiro** [comment]
Adding the `"version"` key, disabled the actions from the command palette but the buttons remain visible.

**Comment by maxdeviant** [comment]
> Adding the `"version"` key disabled the actions from the command palette, but the buttons remain visible.

I played around with this a bit and unfortunately haven't been able to reproduce the problem in Zed v0.129.2.

Does the button visibility persist through a restart of Zed even when `enable` is set to `false`?

**Comment by hferreiro** [comment]
> Does the button visibility persist through a restart of Zed even when `enable` is set to `false`?

Yes.

**Pull request merged**